src/libraries/System.Text.Json/tests/Serialization/ReferenceHandlingTests.Serialize.cs

```diff
@@ -217,7 +217,7 @@ public static void CustomHashCode()
             };
 
             string json = JsonSerializer.Serialize(list, s_serializerOptionsPreserve);
-            Assert.Equal(@"{""$id"":""1"",""$values"":[{""$id"":""2""},{""$id"":""3""}]}", json);
+            Assert.Equal(@"{""$id"":""1"",""$values"":[{""$id"":""2""},{""$ref"":""2""}]}", json);
 
             List<ClassIncorrectHashCode> listCopy = JsonSerializer.Deserialize<List<ClassIncorrectHashCode>>(json, s_serializerOptionsPreserve);
             // When a GetHashCode method is implemented incorrectly, round-tripping breaks,
```

Comment:
Can you reword this and next line coment so it states the new behavior, which is that we will never call user's custom `GetHashCode` implementation.

Reply:
Should I change Assert.NotSame to Assert.Same or add something like `Assert.Equal(RuntimeHelpers.GetHashCode(listCopy[0]), listCopy[0].GetHashCode());`?

Reply:
Yes, please.

Reply:
Thanks for the explanation